registry/extension/oras/referrers.go

```diff
@@ -2,25 +2,42 @@ package oras
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
+	"strconv"
 
 	"github.com/distribution/distribution/v3"
 	dcontext "github.com/distribution/distribution/v3/context"
 	"github.com/distribution/distribution/v3/registry/api/errcode"
 	v2 "github.com/distribution/distribution/v3/registry/api/v2"
+	"github.com/opencontainers/go-digest"
 	orasartifacts "github.com/oras-project/artifacts-spec/specs-go/v1"
 )
 
 // referrersResponse describes the response body of the referrers API.
 type referrersResponse struct {
-	Referrers []orasartifacts.Descriptor `json:"references"`
+	Referrers []orasartifacts.Descriptor `json:"referrers"`
 }
 
+const maxPageSize = 50
+
 func (h *referrersHandler) getReferrers(w http.ResponseWriter, r *http.Request) {
 	dcontext.GetLogger(h.extContext).Debug("Get")
 
 	// This can be empty
 	artifactType := r.FormValue("artifactType")
+	nPage, err := strconv.Atoi(r.FormValue("n"))
+	if nPage < 0 || err != nil {
+		nPage = maxPageSize
+	}
+	nextToken := r.FormValue("nextToken")
+	if nextToken != "" {
+		_, err = digest.Parse(nextToken)
+		if err != nil {
+			h.extContext.Errors = append(h.extContext.Errors, v2.ErrorCodeDigestInvalid.WithDetail("nextToken digest parsing failed"))
```

Comment:
Since the nextToken is opaque to the client, this error type might be confusing. Instead, you could add a new error (such as `ErrorMalformedNextToken`).